### PR TITLE
Bug 1145072 - Implement a queue for key input actions.

### PR DESCRIPTION
**Pull request by RudyLu**

### CODE REVIEW

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=9da2e7fd2c75a074e63ee91363f19f5509b6ca07

File: apps/keyboard/js/keyboard/target_handlers_manager.js [hunk 1]
```diff
@@ -105,7 +106,12 @@ TargetHandlersManager.prototype._callTargetAction = function(action,
     }
   }
 
-  handler[action](press);
+  this.promiseQueue = this.promiseQueue.then(function() {
+    handler[action](press);
+  }).catch(function(e) {
+    console.warn('TargetHandlersManager: ' +
+        'Error occurred for ' + action + ': ' + e);
```
Comment (timdream) [inline apps/keyboard/js/keyboard/target_handlers_manager.js]: `': ', e);` So we could see the stack instead of casting the error object to string.

**Comment by timdream** [comment]
Please also chain all the promises in `CompositeTargetHandler#commit`

https://github.com/RudyLu/gaia/blob/keyboard/Bug1145072-keyboard_action_queue/apps/keyboard/js/keyboard/target_handlers.js#L227-L236

**Comment by timdream** [comment]
And https://github.com/RudyLu/gaia/blob/keyboard/Bug1145072-keyboard_action_queue/apps/keyboard/js/keyboard/target_handlers.js#L31

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=c4784427e35a62b15c24780bb707e9304ffa65b3

**Pull request merged**